packages/sanity/src/tasks/src/tasks/components/activity/TasksActivityLog.tsx

```diff
@@ -134,39 +137,75 @@ type Activity =
 export function TasksActivityLog(props: TasksActivityLogProps) {
   const {value, onChange, path} = props
   const currentUser = useCurrentUser()
+  const {title: workspaceTitle, basePath} = useWorkspace()
 
   const {comments, mentionOptions, operation, getComment} = useComments()
   const loading = comments.loading
   const taskComments = comments.data.open
 
+  const handleGetNotificationValue = useCallback(
+    (message: CommentInputProps['value'], commentId: string) => {
+      const studioUrl = new URL(`${window.location.origin}${basePath}/`)
+      studioUrl.searchParams.set('sidebar', 'tasks')
+      studioUrl.searchParams.set('selectedTask', value?._id)
+      studioUrl.searchParams.set('viewMode', 'edit')
+      studioUrl.searchParams.set('commentId', commentId)
+
+      // We need to update the subscribers list in the task, any user mentioned in the comment should be added
+      // as a subscriber.
+      const mentionedUsers = getMentionedUsers(message)
+      const subscribers = Array.from(new Set([...(value.subscribers || []), ...mentionedUsers]))
+      onChange(set(subscribers, ['subscribers']))
```

Comment:
I find it confusing that a function named `handleGet*` also performs a mutation on the task document. Could we consider patching the subscribers array with the mentioned users at the moment a comment is created or updated instead? For example, in the `handleCommentCreate` and `handleCommentReply` like this:

```ts
const handleCommentCreate = useCallback(
  (message: CommentInputProps['value']) => {
    const commentId = uuid()

    const nextComment: CommentCreatePayload = {
      id: commentId,
      type: 'task',
      message,
      parentCommentId: undefined,
      reactions: EMPTY_ARRAY,
      status: 'open',
      threadId: uuid(),
      context: {
        notification: handleGetNotificationValue(message, commentId),
      },
    }

    // 1. Create the comment
    operation.create(nextComment)

    // 2. Get the mentioned users in the message
    const mentionedUsers = getMentionedUsers(message)

    // 3. If there are mentioned users, update `subscribers` in the task document
    if (mentionedUsers.length > 0) {
      const subscribers = [...(value.subscribers || []), ...mentionedUsers]
      const uniqueSubscribers = Array.from(new Set(subscribers))
      onChange(set(uniqueSubscribers, ['subscribers']))
    }
  },
  [handleGetNotificationValue, operation, value.subscribers, onChange],
)
```

Reply:
I could extract that piece. 
we could have the following:

```
  const handleCommentCreate = useCallback(
    (message: CommentInputProps['value']) => {
      const commentId = uuid()
      const notification = handleGetNotificationValue(message, commentId)
      const nextComment: CommentCreatePayload = {
        id: commentId,
        type: 'task',
        message,
        parentCommentId: undefined,
        reactions: EMPTY_ARRAY,
        status: 'open',
        threadId: uuid(),
        context: {notification},
      }

      operation.create(nextComment)
      onChange(set(notification.subscribers, ['subscribers']))
    },
    [operation, handleGetNotificationValue, onChange],
```

Reply:
I think the task document change should happen before the comment is created, as in the future we may want to get the task values directly in the webhook by resolving that reference.